src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -751,3 +902,135 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+
+/* TODO: write comment */
+void
+InstallRouterExecutorShmemHook()
+{
+	prev_shmem_startup_hook = shmem_startup_hook;
+	shmem_startup_hook = RegisterRouterExecutorXactCallbacks;
+}
+
+
+/* TODO: Write comment */
+static void
+RegisterRouterExecutorXactCallbacks()
+{
+	RegisterXactCallback(RouterTransactionCallback, NULL);
```

Comment:
😦 Hoped this implementation (with the `shmem_startup_hook`) avoided the need for that state variable. So you want me to add it back for an `Assert` or something?